invenio_records_resources/services/records/results.py

```diff
@@ -57,6 +57,7 @@ def data(self):
         self._data = self._service.schema.dump(
             self._identity,
             self._record,
+            record=self._record,
```

Comment:
Good catch. Honestly I think a refactor of `MarshmallowServiceSchema` / `FieldPermissionsMixin` (marshmallow-utils) is in order to take a context parameter and have the record itself be taken into account.